doc: update getting started debug tool (#8932)
nrfconnect/sdk-nrf: doc/nrf/gs_testing.rst

Debugging an application
************************

> To debug an application, set up the debug session as described in `Debugging an application`_ in the |nRFVSC| documentation which uses nRF Debug.
> nRF Debug is the default debugger for |nRFVSC| and uses `Microsoft's debug adaptor`_.

Reviewer: ```suggestion
To debug an application, set up the debug session as described in `Debugging an application`_ in the |nRFVSC| documentation.
nRF Debug is the default debugger for |nRFVSC|.
```

To debug firmware running in the secure part, you need to build Trusted Firmware-M with debug symbols enabled and load the symbols during the debugging session.
To build Trusted Firmware-M with debug symbols, set the :kconfig:option:`CONFIG_TFM_CMAKE_BUILD_TYPE_RELWITHDEBINFO` Kconfig option.
To load the Trusted Firmware-M debug symbols in the GDB tool, run the following command and answer ``y`` if prompted by the tool:

.. code-block:: console

   (gdb) add-symbol-file build/tfm/bin/tfm_s.elf

> For more information about the GDB tool, see `Debug tools`_.

Reviewer: We have nothing similar valid for nRF Debug?

Author: No, nRF Debug does this automatically according to @trond-snekvik.

Debug tools
===========

> The main recommended tool for debugging in the |NCS| is `nRF Debug <Debugging an application_>`_ of the |nRFVSC|.

Reviewer: ```suggestion
The main recommended tool for debugging in the |NCS| is `nRF Debug <Debugging an application_>`_ of the |nRFVSC|.
The tool uses `Microsoft's debug adaptor`_ and integrates custom debugging features specific for the |NCS|.
```

Debug tools
===========

> The main recommended tool for debugging in the |NCS| is the |nRFVSC|'s own nRF Debug, which uses `Microsoft's debug adaptor`_.

Reviewer 2: https://nrfconnect.github.io/vscode-nrf-connect/connect/debug_app.html - could add a link to this page?

> The main recommended tool for debugging in the |NCS| is the |nRFVSC|'s own nRF Debug, which uses `Microsoft's debug adaptor`_.

Author: Added

> The main recommended tool for debugging in the |NCS| is the |nRFVSC|'s own `nRF Debug`_, which uses `Microsoft's debug adaptor`_.

Reviewer 2: ```suggestion
The main recommended tool for debugging in the |NCS| is `nRF Debug <Debugging an application_>`_ of the |nRFVSC|, which uses `Microsoft's debug adaptor`_.
```
